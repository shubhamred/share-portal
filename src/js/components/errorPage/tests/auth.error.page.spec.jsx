import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import AuthErrorPage from '../auth.error.page';

configure({ adapter: new Adapter() });

describe('components/Auth Error page', () => {
  it('renders correctly', () => {
    const component = shallow(
      <AuthErrorPage message="Error" />
    );
    const tree = shallowToJson(component);
    expect(tree)
      .toMatchSnapshot();
  });
  it('renders With props', () => {
    const component = shallow(
      <AuthErrorPage message="Error" />
    );
    const tree = shallowToJson(component);
    expect(tree)
      .toMatchSnapshot();
  });
});
