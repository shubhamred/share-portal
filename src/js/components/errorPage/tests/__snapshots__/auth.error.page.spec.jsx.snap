// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Auth Error page renders With props 1`] = `
<Fragment>
  <div
    className="grow"
  >
    <WithStyles(ForwardRef(AppBar))
      className="appBar"
      position="fixed"
    >
      <WithStyles(ForwardRef(Toolbar))>
        <WithStyles(ForwardRef(Grid))
          container={true}
        >
          <WithStyles(ForwardRef(Grid))
            style={
              Object {
                "padding": "8px",
                "paddingLeft": "0px",
                "width": "72px",
              }
            }
          >
            <img
              alt="Klub"
              src="/assets/klub-logo-white.svg"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Typography))
            className="title"
            noWrap={true}
            variant="h5"
          >
            Portal
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            className="logout"
            onClick={[Function]}
          >
            Logout
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Toolbar))>
    </WithStyles(ForwardRef(AppBar))>
  </div>
  <div
    className="errorContainer"
  >
    <h1
      className="errorCode"
    >
      401
    </h1>
    <p
      className="errorMessage"
    >
      You are not authorized to access this page
    </p>
    <p
      className="errorDescription"
    >
      <b>
        Details : 
      </b>
      Error
    </p>
  </div>
</Fragment>
`;

exports[`components/Auth Error page renders correctly 1`] = `
<Fragment>
  <div
    className="grow"
  >
    <WithStyles(ForwardRef(AppBar))
      className="appBar"
      position="fixed"
    >
      <WithStyles(ForwardRef(Toolbar))>
        <WithStyles(ForwardRef(Grid))
          container={true}
        >
          <WithStyles(ForwardRef(Grid))
            style={
              Object {
                "padding": "8px",
                "paddingLeft": "0px",
                "width": "72px",
              }
            }
          >
            <img
              alt="Klub"
              src="/assets/klub-logo-white.svg"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Typography))
            className="title"
            noWrap={true}
            variant="h5"
          >
            Portal
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            className="logout"
            onClick={[Function]}
          >
            Logout
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Toolbar))>
    </WithStyles(ForwardRef(AppBar))>
  </div>
  <div
    className="errorContainer"
  >
    <h1
      className="errorCode"
    >
      401
    </h1>
    <p
      className="errorMessage"
    >
      You are not authorized to access this page
    </p>
    <p
      className="errorDescription"
    >
      <b>
        Details : 
      </b>
      Error
    </p>
  </div>
</Fragment>
`;
